packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -18,12 +18,104 @@ import {
   MongoNavInterface,
   OrganizationInterface,
   ProjectInterface,
+  HostsInterface,
 } from './types';
 
 const ErrorCodeMap: Record<number, ErrorCode> = {
   401: ErrorCode.NO_AUTHORIZATION,
 } as const;
 
+type UseMongoNavData = Pick<
+  MongoNavInterface,
+  | 'mode'
+  | 'activeOrgId'
+  | 'activeProjectId'
+  | 'loadData'
+  | 'dataFixtures'
+  | 'onSuccess'
+  | 'onError'
+> & {
+  hosts: Required<NonNullable<MongoNavInterface['hosts']>>;
+};
+
+function useMongoNavData({
+  hosts,
+  mode,
+  activeOrgId,
+  activeProjectId,
+  loadData,
+  dataFixtures,
+  onSuccess,
+  onError,
+}: UseMongoNavData): DataInterface | undefined {
+  const [data, setData] = useState<DataInterface | undefined>(undefined);
+
+  const endpointURI = `${hosts.cloud}/user/shared`;
+
+  function fetchProductionData(body?: PostBodyInterface) {
+    const configObject: RequestInit = {
+      credentials: 'include',
+      mode: 'cors',
+      method: 'GET',
+    };
+
+    if (body) {
+      Object.assign(configObject, {
+        method: 'POST',
+        body: JSON.stringify(body),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    }
+
+    return fetch(endpointURI, configObject);
+  }
+
+  function getDataFixtures() {
+    return new Promise(resolve => {
```

Comment:
I believe if you do `return new Promise<DataInterface>(resolve => {` here, you can avoid typecasting, and simplify line 104 to:
```js
getDataFixtures().then(setData);
```

Reply:
Nice catch! This was copy pasta but I'll fix this up!